docs/install/python_wheels.md

```diff
@@ -102,11 +102,16 @@ git clone ssh://bbpcode.epfl.ch/user/kumbhar/mpt-headers
 ```
 
 ## macOS wheels
+
 Note that for macOS there is no docker image needed, but all required dependencies must exist.
 In order to have the wheels working on multiple macOS target versions, special consideration must be made for MACOSX_DEPLOYMENT_TARGET.
 
-Taking Azure macOS wheels for example, `readline` has been manually built with `MACOSX_DEPLOYMENT_TARGET=10.9` and stored as secure file on Azure.
-See [packaging/python/Dockerfile](../../packaging/python/Dockerfile) for readline build instructions.
+
+Taking Azure macOS `x86_64` wheels for example, `readline` was built with `MACOSX_DEPLOYMENT_TARGET=10.9` and stored as secure file on Azure.
+For `arm64`, the wheels currently need to be built manually; in this case we require `MACOSX_DEPLOYMENT_TARGET=11.0` since we are using `universal2` Python installers.
```

Comment:
```11.0` since we are using `universal2` Python installers```
is this true. From brew, I think python is arm64 only. From python.org, they are universal2 (arm64, x86_64) but the ```MACOSX_DEPLOYMENT_TARGET=10.9``` (or am I now out of date in terms of that information).

Reply:
Right. We are using python.org `universal2` installers. I was thrown off by this: 
```
bbptemp@bb-fvff266jq05n Downloads % python3.8 -c "import sysconfig;print(sysconfig.get_platform());"
macosx-11-universal2
```
but
```
bbptemp@bb-fvff266jq05n Downloads % python3.10 -c "import sysconfig;print(sysconfig.get_platform());"
macosx-10.9-universal2
bbptemp@bb-fvff266jq05n Downloads % python3.9 -c "import sysconfig;print(sysconfig.get_platform());"
macosx-10.9-universal2
```

Will check what was installed on the system

Reply:
From running the installers 3.8 vs 3.9, it looks that `universal2` true support comes in from 3.9 only:

3.8:
>This package will install Python 3.8.10 for macOS 11 or later for the following architecture(s): arm64, x86_64.
macOS 11 (Big Sur) and Apple Silicon Mac support [new in 3.8.10]
As of Python 3.8.10, Python is now supported on macOS 11 Big Sur.  The binaries included in this installer install on macOS releases from macOS 10.9 through macOS 11 Big Sur and will run on Apple Silicon Macs using Rosetta 2 translation.  Starting with Python 3.9.1, we provide a new universal2 installer variant which includes binaries that run natively on both Apple Silicon Macs and Intel-based Macs. 

3.9:
>This package will install Python 3.9.10 for macOS 10.9 or later for the following architecture(s): arm64, x86_64.

I need to wrap my head around all this ...

Reply:
> running the installers 3.8

The python you ran seems like the default python3.8.10 from apple. I believe the python.org version is
```
% python3.8
Python 3.8.12 (default, Jan 16 2022, 17:00:20) 
[Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sysconfig
>>> sysconfig.get_platform()
'macosx-10.9-universal2'
```

Reply:
> The python you ran seems like the default python3.8.10 from apple.

3.8.10 is the last official version of 3.8 from python.org.  Maybe you got `3.8.12` from brew?

Reply:
for 3.8.12 we only have the sources: https://www.python.org/ftp/python/3.8.12/

Reply:
Ahh yes. Sorry. Now I remember I had to build from sources for the universal2 and to be consistent with the other
pythons I was using.
```
export MACOS_DEPLOYMENT_TARGET=10.9
./configure  '--prefix=/Users/hines/soft/python3.8' '--enable-universalsdk' '--with-universal-archs=universal2' '--enable-shared' '--with-openssl=/opt/homebrew/opt/openssl' '--with-ssl-default-suites=openssl'
```

Reply:
Yes, brew doesn't provide `universal2` for 3.8. Just checked on both Intel and M1.

Reply:
Ok starting to understand more about this: https://github.com/pypa/cibuildwheel/issues/954](https://github.com/pypa/cibuildwheel/issues/954)

I fear we may have to devise (or not) a strategy for python 3.8 under the risk of not having portable wheels for it.

Reply:
In regard to python macosx target at python.org. If one clicks on Downloads and then clicks on one of the "Looking for a specific release?" group. For those, all the ```macOS 64-bit universal2 installer``` (despite a description ```for macOS 10.9 or later```) have a name like ```python-...-macos11.pkg```.
However, when you open those installers,  you see, for example, the message ```This package will install Python3.9.10 for macOS 10.9 or later.```. And when installed, ```sysconfig.get_platform()``` returns ```macosx-10.9-universal2```.

Reply:
In regard to python macosx target at python.org. If one clicks on Downloads and then clicks on one of the "Looking for a specific release?" group. For those, all the ```macOS 64-bit universal2 installer``` (despite a description ```for macOS 10.9 or later```) have a name like ```python-...-macos11.pkg```.
However, when you open those installers,  you see, for example, the message ```This package will install Python3.9.10 for macOS 10.9 or later.```. And when installed, ```sysconfig.get_platform()``` returns ```macosx-10.9-universal2```.

Reply:
Quote from `pypa issue  #935` mentioned earlier in this thread:
>The problem is that python-3.8.10-macos11.pkg is actually compiled with 11, even for Intel. While for 3.9 and 3.10, the -macos11.pkg packages are properly compiled with 11 for Apple Silicon and 10.9 for Intel, allowing those to be used to make 10.9+ wheels on Intel. You can't do that with the 3.8 universal package, you'd have to remove support for macOS 10 as a target for wheels

We could consider providing `arm64 + macosx_11 + python 3.8` with official `3.8.10` installer.
While it would be possible to use `3.8.12` (that we would have to build ourselves) to provide `arm64 + macosx_10.9 + python 3.8`, I fear that may prove to be  a bad choice in terms of portability of the wheel (i.e. vast majority of users will have `3.8.10`). 

In the near future, when it comes to `universal2` wheels for both `x64_86` and `arm64`, we will do only 3.9+.

Reply:
From https://github.com/pypa/cibuildwheel/issues/954
> While for 3.9 and 3.10, the -macos11.pkg packages are properly compiled with 11 for Apple Silicon and 10.9 for Intel

I'd really like to know how that was done.  All I'm familiar with is the ```MACOSX_DEPLOYMENT_TARGET``` and I thought that was for both Intel and Apple Silicon. (Though I can imagine completely separate builds for x86_64 and arm64 with different MACOSX_DEPLOYMENT_TARGET and then use lipo to mash them together). I'm not sure that the above 
> 11 for Apple Silicon

is accurate since ```sysconfig.get_platform()``` returns ```'macosx-10.9-universal2'``` on my arm64.  But I'm probably misinterpreting something.

Reply:
With python.org official 3.8.10 installer I get: 
```
bbptemp@bb-fvff266jq05n Downloads % python3.8 -c "import sysconfig;print(sysconfig.get_platform());"
macosx-11-universal2
```

Reply:
I get that as well for python3.8 (installed by apple). That is why I built my own python3.8 universal2 with MACOS_DEPLOYMENT_TARGET=10.9

https://github.com/neuronsimulator/nrn/pull/1570#issuecomment-1013968345
details the shenanigans I went through.

Reply:
Indeed, that's the main problem with 3.8, there's nothing official with 10.9.